paimon-core/src/main/java/org/apache/paimon/metrics/commit/CommitStats.java

```diff
@@ -0,0 +1,376 @@
+        partitionedChangelogFilesAppended.forEach(
+                (k, v) ->
+                        this.partitionedNumChangelogRecordsAppended.put(
+                                k, v.stream().mapToLong(file -> file.rowCount()).sum()));
+
+        bucketedTableFilesCompacted.forEach(
+                (k, v) ->
+                        this.bucketedNumDeltaRecordsCompacted.put(
+                                k, v.stream().mapToLong(file -> file.rowCount()).sum()));
+        partitionedTableFilesCompacted.forEach(
+                (k, v) ->
+                        this.partitionedNumDeltaRecordsCompacted.put(
+                                k, v.stream().mapToLong(file -> file.rowCount()).sum()));
+
+        bucketedChangelogFilesCompacted.forEach(
+                (k, v) ->
+                        this.bucketedNumChangelogRecordsCompacted.put(
+                                k, v.stream().mapToLong(file -> file.rowCount()).sum()));
+        partitionedChangelogFilesCompacted.forEach(
+                (k, v) ->
+                        this.partitionedNumChangelogRecordsCompacted.put(
+                                k, v.stream().mapToLong(file -> file.rowCount()).sum()));
+
+        this.tableFilesAdded = addedTableFiles.size();
+        this.tableFilesDeleted = deletedTableFiles.size();
+        this.changelogFilesCommitAppended = appendChangelogFiles.size();
+        this.changelogFilesCompacted = compactChangelogFiles.size();
+        this.numPartitionsWritten = numChangedPartitions(appendTableFiles, compactTableFiles);
+        this.numBucketsWritten = numChangedBuckets(appendTableFiles, compactTableFiles);
+        this.partitionsWritten = changedPartitions(appendTableFiles, compactTableFiles);
+        this.bucketsWritten = changedBuckets(appendTableFiles, compactTableFiles);
+        this.duration = commitDuration;
+        this.generatedSnapshots = generatedSnapshots;
+        this.attempts = attempts;
+    }
+
+    /** Return a map group by partition. */
+    @VisibleForTesting
+    protected static Map<BinaryRow, List<DataFileMeta>> groupByPartititon(
+            List<ManifestEntry> files) {
+        Map<BinaryRow, List<DataFileMeta>> groupByPartition = new LinkedHashMap<>();
+        for (ManifestEntry entry : files) {
+            groupByPartition
+                    .computeIfAbsent(entry.partition(), k -> new ArrayList<>())
+                    .add(entry.file());
+        }
+        return groupByPartition;
+    }
+
+    /** Return a map group by bucket. */
+    @VisibleForTesting
+    protected static Map<Integer, List<DataFileMeta>> groupByBucket(List<ManifestEntry> files) {
+        Map<Integer, List<DataFileMeta>> groupByBucket = new LinkedHashMap<>();
+        for (ManifestEntry entry : files) {
+            groupByBucket.computeIfAbsent(entry.bucket(), k -> new ArrayList<>()).add(entry.file());
+        }
+        return groupByBucket;
+    }
```

Comment:
It is not very useful to only group by bucket and ignores partition. Bucket 1 in partition 1 is different from bucket 1 in partition 2, although they have the same bucket number.

Reply:
Yes sure, I understand the scenario. Updated the group design.